Sources/Honeycomb/HoneycombInstrumentedView.swift

```diff
@@ -0,0 +1,64 @@
+import OpenTelemetryApi
+import SwiftUI
+
+private let honeycombInstrumentedViewName = "@honeycombio/instrumentation-view"
+
+struct HoneycombInstrumentedView<Content: View>: View {
+    private let span: Span
+    private let content: () -> Content
+    private let name: String
+    private let initTime: Date
+
+    init(name: String, @ViewBuilder _ content: @escaping () -> Content) {
+        self.initTime = Date()
+        self.name = name
+        self.content = content
+
+        self.span = getViewTracer().spanBuilder(spanName: "View Render")
+            .setStartTime(time: Date())
+            .setAttribute(key: "ViewName", value: name)
+            .startSpan()
+
+        print("\(name) init")
+    }
+
+    var body: some View {
+        print("\(name) body started rendering")
+        let start = Date()
+
+        // contents start init
+        let bodySpan = getViewTracer().spanBuilder(spanName: "View Body")
+            .setStartTime(time: Date())
+            .setAttribute(key: "ViewName", value: name)
+            .setParent(span)
+            .setActive(true)
+            .startSpan()
+
+        let c = content()
+
+        let endTime = Date()
+        // contents end init
+
+        print("\(name) body finished rendering")
+        span.setAttribute(
+            key: "RenderDuration",
+            value: endTime.timeIntervalSince(start)
+        )
+
+        return c.onAppear {
+            // contents end render
+
+            print("\(name) content appeared")
+            span.setAttribute(key: "TotalDuration", value: Date().timeIntervalSince(initTime))
+            span.end(time: Date())
```

Comment:
Would it be better to capture the `Date()` once and use the same value on both these lines?